arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -2058,11 +2058,6 @@ Result RocksDBEngine::dropCollection(TRI_vocbase_t& vocbase,
   bool const prefixSameAsStart = true;
   bool const useRangeDelete = rcoll->meta().numberDocuments() >= 32 * 1024;
 
-  auto resLock = rcoll->lockWrite().get();  // technically not necessary
```

Comment:
While the comments mentions that this is "technically not necessary", acquiring the lock here may previously have led to synchronization with all ongoing write operations to the collection being fully finished.
With the lock being removed here now, that behavior may or may not change. I can't tell.
To be on the safe side, we could add the lock back even if we can't prove it is actually needed. Or we try and see.